ado/ipatrackversions.ado

```diff
@@ -0,0 +1,195 @@
+*! version 1.0.0 Caton brewster 10nov2016
+
+program ipatrackversions, rclass
+	/* Create a sheet that shows the survey versions used for each day of surveying. 
+	For the most recent day of surveying, if there are surveys that have been submitted
+	using the wrong version, list the enumerator ID, respondent ID, etc. for those observations
+	to facilitate finding that enumerator and ensuring they upload the latest version of the 
+	survey */
+	
+	
+
+//define inputs
+version 13
+
+#delimit ;
+syntax varname,  //varname is the form version variable - must be num. 
+	/* specify uid for the survey, enumerator id */
+    id(varname) ENUMerator(varname) 
+	/* list of other vars to keep */
+	[KEEPvars(string)] 
+	/* specify date var, i.e. submission date var */
+	submit(varname numeric)
+	/* output filename */
+	saving(string) 
+	;	
+#delimit cr
+
+	
+	di ""
+	di "Compiling information on survey form versions by submission date..."
+
+	qui {
+	
+	* test for fatal conditions 
+	cap confirm numeric var `varlist'
+	if _rc {
+		di as err "Variable used for form version (`varlist') not numeric. Must be numeric and ordinal for the program to work correctly"
+		error 101
+	}
+		
+	if !mi("`keep'") {
+		foreach var of varlist `keep' {
+			cap confirm var `var'
+			if _rc {
+				di as err `"Tried to specify keeping the var "`var'" - "`var'" does not exist in the dataset"'
+				error 101
+			}
+		}
+	}
+	
+	count if `submit' == . 
+	if `r(N)' > 0 {
+		di as err `"There are missing values of `submit'. Either drop these observations or restrict them using an "if" statement."'
```

Comment:
What do you mean by "restrict them using an `if` statement"? FYI if you want them to be able to use `if` with your command you have to add it to the `syntax` statement.